### PR TITLE
Using clock to allow mocking time

### LINKED ISSUE #27: Use clock as time provider to allow testing the timer
clock (https://pub.dev/packages/clock) + fake_async (https://pub.dev/packages/fake_async) allows mocking of current time. It would be nice if cron could use `clock.now()` instead of `DateTime.now()` to allow the same. 

If you are interested in this but have no available time right now, I can try to implement it myself.

### PR DESCRIPTION
**Pull request by pescuma**
I've added a couple of tests too, just to get started. 

This closes #27.

### CODE REVIEW

File: lib/cron.dart [hunk 1]
```diff
@@ -43,6 +45,15 @@ class Schedule {
   /// The weekdays a Task should be started.
   final List<int>? weekdays;
 
+  bool runAt(DateTime time) {
```
Comment (isoos) [inline lib/cron.dart:48]: Does this needs to be public?

**Review by isoos — COMMENTED** [review]
Looks good overall, only a nit question....

**Comment by isoos** [comment]
Also: could you please bump the version (0.3.3 or 0.4.0) and add a short description to changelog?

File: lib/cron.dart (hunk 1 above)
Comment (pescuma) [reply lib/cron.dart:48]: I'm using this method in my project, so I'd prefer if it were.

**Comment by pescuma** [comment]
Done

File: lib/cron.dart (hunk 1 above)
Comment (isoos) [reply lib/cron.dart:48]: I don't mind having it public, but let's add a small documentation describing what it means.

Also, really nitpicking, but the current name suggest a command: [let's] run at... Instead, maybe `willRunAt`, `shouldRunAt` or `runsAt` seems to be more fitting, which indicates that the method returns with a yes/no (bool) answer. I don't have a strong preference here, but I think a different name would really help the readability.

**Pull request merged**